> @primeos Not sure about the fix for the sway test. I don't see that sway strictly needs polkit, but the tests are not working without it. But maybe it's something in the tests requiring it?

Right, it shouldn't be required for Sway itself. It looks like it might be required for some logind functionality though (doesn't systemd in general rely on Polkit btw? It's a while ago that I peeked into that part but IIRC there are quite some connections between the two, unfortunately (I didn't like it back then either so huge thanks for looking into disabling it!) - ideally Polkit is optional though).
I've dumped Sway's output and got this:
```
00:00:00.043 [DEBUG] [sway/server.c:50] Preparing Wayland server initialization
00:00:00.057 [INFO] [wlr] [libseat] [libseat/backend/seatd.c:80] Could not connect to socket /run/seatd.sock: No such file or directory
00:00:00.058 [INFO] [wlr] [libseat] [libseat/libseat.c:76] Backend 'seatd' failed to open seat, skipping
00:00:00.070 [ERROR] [wlr] [libseat] [libseat/backend/logind.c:310] Could not activate session: Permission denied
00:00:00.071 [INFO] [wlr] [libseat] [libseat/libseat.c:76] Backend 'logind' failed to open seat, skipping
00:00:00.071 [INFO] [wlr] [libseat] [seatd/seat.c:38] Created VT-bound seat seat0
00:00:00.072 [INFO] [wlr] [libseat] [seatd/server.c:139] New client connected (pid: 889, uid: 1000, gid: 100)
00:00:00.072 [ERROR] [wlr] [libseat] [common/terminal.c:149] Could not open target tty: Permission denied
00:00:00.073 [ERROR] [wlr] [libseat] [seatd/seat.c:60] Could not open tty0 to update VT: Permission denied
00:00:00.073 [INFO] [wlr] [libseat] [seatd/seat.c:169] Added client 0 to seat0
00:00:00.073 [ERROR] [wlr] [libseat] [common/terminal.c:149] Could not open target tty: Permission denied
00:00:00.073 [ERROR] [wlr] [libseat] [seatd/seat.c:71] Could not open terminal for VT 0: Permission denied
00:00:00.073 [ERROR] [wlr] [libseat] [seatd/seat.c:449] Could not open VT for client
00:00:00.074 [ERROR] [wlr] [libseat] [common/terminal.c:149] Could not open target tty: Permission denied
00:00:00.074 [ERROR] [wlr] [libseat] [seatd/seat.c:85] Could not open terminal to clean up VT 0: Permission denied
00:00:00.074 [INFO] [wlr] [libseat] [libseat/libseat.c:73] Seat opened with backend 'builtin'
00:00:00.074 [INFO] [wlr] [backend/session/session.c:110] Successfully loaded libseat session
00:00:00.074 [INFO] [wlr] [backend/backend.c:91] Waiting for a session to become active
00:00:10.085 [ERROR] [wlr] [backend/backend.c:114] Timeout waiting session to become active
00:00:10.085 [ERROR] [wlr] [backend/backend.c:352] Failed to start a DRM session
00:00:10.085 [ERROR] [sway/server.c:56] Unable to create backend
00:00:10.095 [ERROR] [wlr] [libseat] [common/terminal.c:149] Could not open target tty: Permission denied
00:00:10.095 [ERROR] [wlr] [libseat] [seatd/seat.c:85] Could not open terminal to clean up VT 0: Permission denied
00:00:10.095 [INFO] [wlr] [libseat] [seatd/seat.c:512] Closed client 0 on seat0
00:00:10.095 [INFO] [wlr] [libseat] [seatd/seat.c:191] Removed client 0 from seat0
```

The problem should be:
```
00:00:00.070 [ERROR] [wlr] [libseat] [libseat/backend/logind.c:310] Could not activate session: Permission denied
```

(`Could not connect to socket /run/seatd.sock` is to be expected.)

So the problem shouldn't be specific to Sway (dependency wise it's quite nested: sway -> wlroots -> seatd -> systemd/logind).
Unfortunately I haven't time for a closer look though :o

--- a/nixos/doc/manual/from_md/release-notes/rl-2205.section.xml
+++ b/nixos/doc/manual/from_md/release-notes/rl-2205.section.xml
@@ -721,6 +721,14 @@
           <literal>tmux</literal>.
         </para>
       </listitem>
+      <listitem>
+        <para>
+          The polkit service, available at
+          <literal>security.polkit.enable</literal>, is now disabled by
+          default. It will automatically be enabled through services and
+          desktop environments as needed.
+        </para>
+      </listitem>
     </itemizedlist>
   </section>
 </section>

--- a/nixos/doc/manual/release-notes/rl-2205.section.md
+++ b/nixos/doc/manual/release-notes/rl-2205.section.md
@@ -245,4 +245,6 @@ In addition to numerous new and upgraded packages, this release has the followin
 
 - `programs.tmux` has a new option `plugins` that accepts a list of packages from the `tmuxPlugins` group. The specified packages are added to the system and loaded by `tmux`.
 
+- The polkit service, available at `security.polkit.enable`, is now disabled by default. It will automatically be enabled through services and desktop environments as needed.
+
 <!-- To avoid merge conflicts, consider adding your item at an arbitrary place in the list instead. -->

--- a/nixos/modules/programs/sway.nix
+++ b/nixos/modules/programs/sway.nix
@@ -134,6 +134,7 @@ in {
         '';
       };
     };
+    security.polkit.enable = true;
     security.pam.services.swaylock = {};
     hardware.opengl.enable = mkDefault true;
     fonts.enableDefaultFonts = mkDefault true;

--- a/nixos/modules/security/polkit.nix
+++ b/nixos/modules/security/polkit.nix
@@ -12,11 +12,7 @@ in
 
   options = {
 
-    security.polkit.enable = mkOption {
-      type = types.bool;
-      default = true;
-      description = "Whether to enable PolKit.";
-    };
+    security.polkit.enable = mkEnableOption "polkit";
 
     security.polkit.extraConfig = mkOption {
       type = types.lines;

--- a/nixos/modules/services/wayland/cage.nix
+++ b/nixos/modules/services/wayland/cage.nix
@@ -81,6 +81,8 @@ in {
       };
     };
 
+    security.polkit.enable = true;
+
     security.pam.services.cage.text = ''
       auth    required pam_unix.so nullok
       account required pam_unix.so

